packages/api-explorer/src/form-components/MultiSchemaField.jsx

```diff
@@ -0,0 +1,118 @@
+require('../../style/main.scss');
+
+const React = require('react');
+const PropTypes = require('prop-types');
+
+const BaseMultiSchemaField = require('@readme/oas-form/src/components/fields/MultiSchemaField').default;
+const fieldProps = require('@readme/oas-form/src/types').fieldProps;
+
+function extractDiscriminatorField(schema, property) {
+  if (schema.properties?.[property]) {
+    const discSchema = schema.properties[property];
+    delete schema.properties[property];
+    return discSchema;
+  }
+
+  // Sometimes discriminators are within each of the oneOf schemas, and not top level.
+  if (schema.oneOf) {
+    return extractDiscriminatorField(schema.oneOf[0], property);
```

Comment:
It possible for the discriminator to be inside just one of the entries in a multi-array `oneOf`?

Reply:
When the discriminator is outside of the `oneOf`, it means that property MUST exist within the response payload, i.e. in each of the `oneOf`s. The behavior of discriminators within each of the oneOf schemas isn't clearly defined or regulated, but I would assume it works similarly. I haven't found any examples otherwise.

Reply:
👍  Doesn't look like the JSON Schema that the OAI offers validates that, but is probably a safe assumption for it to be in each. https://github.com/OAI/OpenAPI-Specification/blob/master/schemas/v3.0/schema.json